### ISSUE TITLE
Drift in state for appsec resources

### ISSUE THREAD

**Issue by partamonov** [akamai/terraform-provider-akamai#228]
### Terraform Version

```shell
terraform version
Terraform v0.14.8
+ provider registry.terraform.io/akamai/akamai v1.5.0 (also tried with 1.4.x)
+ provider registry.terraform.io/hashicorp/aws v3.37.0
```

### Affected Resource(s)

- akamai_appsec_advanced_settings_logging
- akamai_appsec_custom_rule

### Terraform Configuration Files

```hcl
resource "akamai_appsec_advanced_settings_logging" "logging" {
  config_id = local.appsec_config_id
  version   = akamai_appsec_configuration_version_clone.clone.version
  // Also tried with json from file - result is same
  logging = jsonencode(
    {
      "allowSampling" = true,
      "cookies" = {
        "type" = "none"
      },
      "customHeaders" = {
        "type" = "all"
      },
      "override" = false,
      "standardHeaders" = {
        "type" = "all"
      }
    }
  )
}

resource "akamai_appsec_custom_rule" "tls_visibility" {
  config_id = local.appsec_config_id
  custom_rule = jsonencode(
    {
      conditions = [
        {
          positiveMatch = true
          type          = "hostMatch"
          value = [
            "www.example.com",
            "example.com",
          ]
        },
        {
          positiveMatch = true
          type          = "pathMatch"
          value = [
            "/en/retail*",
            "/fr/retail*",
            "/nl/retail*",
          ]
        },
      ]
      description   = ""
      name          = "TLS hash visibility rule"
      tag = [
        "tls",
      ]
    }
  )
}
```

### Expected Behavior

After apply there should not be any changes in the next run

### Actual Behavior

Every run I see that resource will be updated in place

```
# akamai_appsec_advanced_settings_logging.logging will be updated in-place
  ~ resource "akamai_appsec_advanced_settings_logging" "logging" {
        id        = "12345:12:"
      ~ logging   = jsonencode( # whitespace changes
            {
                allowSampling   = true
                cookies         = {
                    type = "none"
                }
                customHeaders   = {
                    type = "all"
                }
                override        = false
                standardHeaders = {
                    type = "all"
                }
            }
        )
        # (2 unchanged attributes hidden)
    }

~ resource "akamai_appsec_custom_rule" "tls_visibility" {
      ~ custom_rule    = jsonencode(
          ~ {
              - id            = 12345678 -> null
              - ruleActivated = false -> null
              - structured    = true -> null
              - version       = 4 -> null
                # (4 unchanged elements hidden)
            }
        )
        id             = "12345:12345678"
        # (2 unchanged attributes hidden)
    }
```

In case of `akamai_appsec_advanced_settings_logging` some whitespaces are causing issues and if value can be saved in state file as minified/normalized json string, then it can solve issue

In case of `akamai_appsec_custom_rule` looks like read-only values should be ignored and not placed in state file or should be taken out from `custom_rule` parameter and be as first-class parameter in resource and `custom_rule` renamed to `custom_conditions`

What I mean is

```hcl
resource "akamai_appsec_custom_rule" "tls_visibility" {
  config_id = local.appsec_config_id
  description   = ""
  name          = "TLS hash visibility rule"
  tag = [ "tls"]
  id = (would be computed value and set by d.SetId() call)
  rule_activated = (would be computed value)
  structured = (would be computed value)
  version = (would be computed value)
  conditions = jsonencode(
    [
        {
          positiveMatch = true
          type          = "hostMatch"
          value = [
            "www.example.com",
            "example.com",
          ]
        },
        {
          positiveMatch = true
          type          = "pathMatch"
          value = [
            "/en/retail*",
            "/fr/retail*",
            "/nl/retail*",
          ]
        },
      ]
  )
}
```

### Steps to Reproduce

1. `terraform apply`
2. `terraform plan`
3. `terraform apply`

**Comment by atuljain16** [comment]
@partamonov This has been addressed in the current release 1.5.1, that went live today.

**Comment by partamonov** [comment]
Confirm both cases are fixed in 1.5.1
